static/js/ducks/sources.js

```diff
@@ -4,14 +4,17 @@ import store from '../store';
 
 export const FETCH_SOURCES = 'skyportal/FETCH_SOURCES';
 export const FETCH_SOURCES_OK = 'skyportal/FETCH_SOURCES_OK';
+export const FETCH_SOURCES_FAIL = 'skyportal/FETCH_SOURCES_FAIL';
 
 
-export function fetchSources(page=1) {
-  return API.GET(`/api/sources?page=${page}`, FETCH_SOURCES);
-}
-
-export function submitSourceFilterParams(formData) {
-  return API.POST(`/api/sources/filter`, FETCH_SOURCES, formData);
+export function fetchSources(filterParams={}) {
+  if (!Object.keys(filterParams).includes("pageNumber")) {
+    filterParams.pageNumber = 1;
+  }
+  const queryString = Object.keys(filterParams).map(
```

Comment:
I'm not sure this is safe; shouldn't there be some form of escaping?  Perhaps fetch already supports providing the query arguments in another way.

Reply:
Fetch doesn't provide that (the docs say to just put query parameters in the URL). Do you mean just URI encoding certain characters? Sure, I'll add this, which does the mapping and URI encodes any special characters: https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams/toString

Reply:
@stefanv I've updated this, take a look now. All relevant characters are now URI-encoded and all URL query strings are safe and to spec. Should be good to go.